system-contract-dapp-playground/src/components/contract-interaction/erc/erc-20/common/MultiLineMethod.tsx

```diff
@@ -0,0 +1,107 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { Input, Tooltip } from '@chakra-ui/react';
+
+interface PageProps {
+  paramFields: {
+    title: string;
+    inputType: string;
+    inputPlaceholder: string;
+    inputSize: string;
+    inputFocusBorderColor: string;
+    inputClassname: string;
+  }[];
+  methodName: string;
+  params: any;
+  setParams: any;
+  explanation: string;
+}
+
+const MultiLineMethod = ({
+  paramFields,
+  params,
+  setParams,
+  methodName,
+  explanation,
+}: PageProps) => {
+  return (
+    <div className="flex flex-col w-[360px] gap-6">
+      {/* inputs */}
+      <div className="flex flex-col gap-3">
+        {/* spender */}
+        {paramFields.map((paramField) => {
+          let value = '';
+          let handleOnInputChange;
+          if (paramField.title === 'spenderAddress') {
```

Comment:
Q: instead of having all of these if statements is it possible to pull the param to set.
It would be clearer for `MultilIneMethod.tsx` to not have to be aware of what field it contains but rather just take a list of fields to display

Reply:
Right that's indeed the goal which I have been doing the same for other components but this is a different case. The main key point is the `handleOnInputChange` method. I used only 1 `<Input />` component for a list of different input fields like `spender address` or `owner address` etc., and the `setParams` is also a sort of like wrapper function for different input fields, say `transfer` params for example, it has this 
```
const [transferParams, setTransferParams] = useState({
    recipient: '',
    amount: '',
  });
  ```
 while `transferFrom` has this 
 ```
 const [transferFromParams, setTransferFromParams] = useState({
    owner: '',
    recipient: '',
    amount: '',
  });
  ```
These are two different object structures, and go on there are going to be more different APIs with different methods with different param structures. So it might look lengthy for `MultilIneMethod.tsx` but this current setup is for the `setParams` to correctly and dynamically update its input states. One way to make all things cleaner is using `React/Redux` state management library. But setting this up can take a quite some time.

Reply:
On a side, the `handleOnInputChange` is misspelled. Pushing new commit to fix it to `handleInputOnChange`

Reply:
I still think a cleaner approach would be one in where you pass it what to display and it doesn't have to filter anything